hikari/internal/aio.py

```diff
@@ -36,6 +36,10 @@
 import inspect
 import typing
 
+if typing.TYPE_CHECKING:
+    # typing_extensions is a dependency of mypy, and pyright vendors it.
```

Comment:
`mypy` is not installed as a requirement as hikari, so this will break if people try to analyze typehints without mypy installed

Reply:
I think this is probably fine -- if anyone complains, it can easily be fixed then. Adding a runtime dependency for something that is supposed to be static just doesn't sit right with me...

Additionally, this will only error if forward annotations suddenly die, or if someone runs `typing.get_type_hints` on the functions in question: both are not very likely imo, especially this is internal to the library.

Reply:
@FasterSpeeding ?

Reply:
Seems fine to me